appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -108,22 +146,41 @@ internal class ExperienceRenderer(
 
         val experience = experiences.first()
         if (!show(experience)) {
-            analyticsTracker.trackExperienceError(experience)
-
             attemptToShow(experiences.drop(1))
         } else {
             analyticsTracker.trackExperienceRecovery(experience)
         }
     }
 
-    suspend fun startFrame(frameId: String, frame: AppcuesFrameView) {
-        renderContextManager.registerEmbedFrame(frameId, frame)
+    suspend fun start(owner: StateMachineOwning, context: RenderContext) {
+        // If there's already a frame for the context, reset it back to its unregistered state.
+        val existingOwner = stateMachines.getOwner(context) as? AppcuesFrameStateMachineOwner
+        if (existingOwner != null) {
+            existingOwner.frame.get()?.reset()
+            existingOwner.stateMachine = null
+        }
 
-        show(RenderContext.Embed(frameId))
+        // If the machine being started is already registered for a different context,
+        // reset it back to its unregistered state before potentially showing new content.
+        if (owner.stateMachine != null) {
+            (owner as? AppcuesFrameStateMachineOwner)?.let {
+                it.frame.get()?.reset()
+                it.stateMachine = null
+            }
+        }
+
+        owner.stateMachine = get<StateMachine>().also {
+            lifecycleTracker.start(it, { potentiallyRenderableExperiences.remove(renderContext) })
+        }
+        stateMachines.setOwner(context, owner)
```

Comment:
this is where we create and register the StateMachine just once, when the frame is registered - rather than on demand - so we can know if a frame should be valid or not